src/util/psbt/mod.rs

```diff
@@ -0,0 +1,548 @@
+// Rust Bitcoin Library
+// Written by
+//   The Rust Bitcoin developers
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! # Partially Signed Transactions
+//!
+//! Implementation of BIP174 Partially Signed Bitcoin Transaction Format as
+//! defined at https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki
+//! except we define PSBTs containing non-standard SigHash types as invalid.
+
+use blockdata::transaction::Transaction;
+use consensus::encode::{self, Encodable, Decodable, Encoder, Decoder};
+
+mod error;
+pub use self::error::Error;
+
+pub mod raw;
+
+#[macro_use]
+mod macros;
+
+pub mod serialize;
+
+mod map;
+pub use self::map::{Map, Global, Input, Output};
+
+/// A Partially Signed Transaction.
+#[derive(Debug, Clone, PartialEq)]
+pub struct PartiallySignedTransaction {
+    /// The key-value pairs for all global data.
+    global: Global,
```

Comment:
These variables being private makes it almost impossible to use this. Is that a mistake?